src/AdvancedADC.cpp

```diff
@@ -115,42 +115,73 @@ DMABuffer<Sample> &AdvancedADC::read() {
     return NULLBUF;
 }
 
-int AdvancedADC::begin(uint32_t resolution, uint32_t sample_rate, size_t n_samples, size_t n_buffers) {
+int AdvancedADC::begin(uint32_t resolution, uint32_t sample_rate, size_t n_samples, size_t n_buffers,bool do_start,uint8_t adcNum) {
+    
     ADCName instance = ADC_NP;
-
     // Sanity checks.
     if (resolution >= AN_ARRAY_SIZE(ADC_RES_LUT) || (descr && descr->pool)) {
         return 0;
     }
 
+    //if ADC specified is more than the number of available ADC bail out
+    if(adcNum>AN_ARRAY_SIZE(adc_pin_alt))
+    {
+        descr = nullptr;
+        return 0;
+    }
```

Comment:
```suggestion
    if (adcNum > AN_ARRAY_SIZE(adc_pin_alt)) {
        descr = nullptr;
        return 0;
    }
```

Spaces between operators, bracket on the same line as function/if.